### PR TITLE
perf: Enable CA1810 performance rule

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Performance

## What is the current behavior?

CA1810 analyzer not enabled.

## What is the new behavior?

Enabled and fixed violations.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Related to https://github.com/unoplatform/uno/pull/6892#issuecomment-915489031.

Internal Issue (If applicable):

### CODE REVIEW

**Review by Youssef1313 — COMMENTED** [review]
This needs a very careful review :)

File: src/Uno.UWP/System/Power/PowerManager.iOS.cs [hunk 1]
```diff
@@ -17,17 +17,17 @@ public partial class PowerManager
...
 		{
-			_device = UIDevice.CurrentDevice;
-			_isSimulator = _device.Model?
+			var isSimulator = _device.Model?
 				               .Contains(
 					               DeviceModelSimulator,
 					               StringComparison.InvariantCultureIgnoreCase) == true;
-			_device.BatteryMonitoringEnabled = !_isSimulator;
+			_device.BatteryMonitoringEnabled = !isSimulator;
+			return isSimulator;
```
Comment (jeromelaban) [inline src/Uno.UWP/System/Power/PowerManager.iOS.cs:30]: This fees like it's assuming with the order of assignation of the `_device` static field and could hide subtle bugs.

File: src/Uno.UWP/UI/Core/CoreDispatcher.Android.cs [hunk 2]
```diff
@@ -128,16 +128,10 @@ private void DispatchItemsToChoreographer()
 		[Register("mono/java/lang/CoreDispatcherImplementor")]
 		internal sealed class CoreDispatcherImplementor : Java.Lang.Object, IRunnable
 		{
-			private readonly static IntPtr _class;
-			private readonly static IntPtr _ctor;
+			private readonly static IntPtr _class = JNIEnv.FindClass(typeof(CoreDispatcherImplementor));
+			private readonly static IntPtr _ctor = JNIEnv.GetMethodID(_class, "<init>", "()V");
```
Comment (jeromelaban) [inline src/Uno.UWP/UI/Core/CoreDispatcher.Android.cs:132]: Same here, the order of execution may be tricky.

**Review by jeromelaban — COMMENTED** [review]
Implied order of execution is not particulatly fun to debug on devices... I'd rather have the mentioned blocks ignored for this rule.

File: src/Uno.UI/Controls/NSFontHelper.macOS.cs [hunk 3]
```diff
@@ -17,17 +17,11 @@ namespace Windows.UI
 {
 	internal static class NSFontHelper
 	{
-		private static Func<nfloat, FontWeight, FontStyle, FontFamily, nfloat?, NSFont> _tryGetFont;
+		private static Func<nfloat, FontWeight, FontStyle, FontFamily, nfloat?, NSFont> _tryGetFont = ((Func<nfloat, FontWeight, FontStyle, FontFamily, nfloat?, NSFont>)InternalTryGetFont).AsMemoized();
```
Comment (jeromelaban) [inline src/Uno.UI/Controls/NSFontHelper.macOS.cs:20]: I think we can use this instead:

```suggestion
		private static Func<nfloat, FontWeight, FontStyle, FontFamily, nfloat?, NSFont> _tryGetFont = Funcs.Create(InternalTryGetFont).AsMemoized();
```

File: src/Uno.UWP/System/Power/PowerManager.iOS.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UWP/System/Power/PowerManager.iOS.cs:30]: The order of execution matches the order of declarations in source code. Do you want me to add a comment stating that order is important, or revert the ones you commented on?

**Comment by jeromelaban** [comment]
@Youssef1313 this PR is very outdated, closing for now. We'll revisit with the latest sources.

**Pull request closed without merging**